### ISSUE TITLE
Will the new booster recommendations from STIKO be reflected in the booster notifications send by CWA?

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-documentation#754]
## Your Question
* Documentation File: https://www.rki.de/DE/Content/Kommissionen/STIKO/Empfehlungen/PM_2021-11-18.html
* Question: Will the changed recommendation form the STIKO, which recommends a booster shot after 6 months for everyone above 18, be reflected in the booster notifications send by the CWA?

If yes: When will this change go live?

**Comment by arnebr** [comment]
Hi, there is no need for an update in the app, since it is done automatically in background via the business rules update from the German gateway. 
The regarding rule is published since 2021-11-19T00:00:00Z.
<img width="1436" alt="Bildschirmfoto 2021-11-25 um 11 56 05" src="https://user-images.githubusercontent.com/1068416/143428913-07f25fdf-73a5-4573-a54e-4c1f5cef7b54.png">

**Comment by dsarkar** [comment]
@Ein-Tim Thanks for the question.
@arnebr Thanks for providing the information.

We have examples from users, that did already receive such a notification.
![image](https://user-images.githubusercontent.com/1409741/143430160-07d52b7c-8bd4-4a58-a4c1-bf5809225b35.png)

Does this answer your question, @Ein-Tim?

**Comment by Ein-Tim** [comment]
@arnebr 

I know that there is no update needed and that the booster rules are downloaded from the gateway. 

The rule you screenshotted is a new rule which is based on the recommendation by the BMG. 
My question, however, was whether the new STIKO recommendations, will also be reflected in the app.

**Comment by Ein-Tim** [comment]
@dsarkar 

No, sorry, it doesn't. The new rules are based on the recommendation from the BMG, but not the incoming change of recommendation from the STIKO.

**Comment by dsarkar** [comment]
@Ein-Tim Alright, got it. So far this is what I can tell you: There are thoughts about to change the text in the notification for the rule 0300. The reason being: Avoid the possible situation with another booster rule, that users receive another notification having received already one. 

When you say "... reflected in the app" do you refer to booster notifications or any other means?

**Comment by Ein-Tim** [comment]
@dsarkar Okay, thanks for the feedback! 

I guess once the STIKO officially changed their recommendation, the BNR will be adapted, right?

> When you say "... reflected in the app" do you refer to booster notifications or any other means?

I only mean the booster notifications (:

**Comment by Ein-Tim** [comment]
@dsarkar So as of today, STIKO also recommends to get bolstered after 3 months. The BNR is still only send after 6 and the text speaks about 6. When will the booster notification rules be updated?

**Comment by MikeMcC399** [comment]
@Ein-Tim 
There is a press-release from STIKO on https://www.rki.de/DE/Content/Kommissionen/STIKO/Empfehlungen/PM_2021-12-21.html from today, so it would make sense for these recommendations to be put into the BNR-DE-* ruleset.

**Comment by MikeMcC399** [comment]
@dsarkar 

Could you check the BNR-DE-0300 rule on https://distribution.dcc-rules.de/bnrules/cd7defef43c2c0edb115090e669c2bb9028608e247c71b3d0f44728a3df4126f please? It looks like this is only triggered for persons older than 70 years old, however the [Epidemiologisches Bulletin 48/2021](https://www.rki.de/DE/Content/Infekt/EpidBull/Archiv/2021/Ausgaben/48_21.pdf?__blob=publicationFile) reduced this to >= 18 years old last month.

In your comment on https://github.com/corona-warn-app/cwa-documentation/issues/754#issuecomment-979097954, you did not mention how old the persons were who received the booster notification.

```
          {
            "plusTime": [
              {
                "var": "payload.dob"
              },
              70,
              "year"
            ]
          }
```

**Comment by dsarkar** [comment]
@MikeMcC399 ok, I will try to get some information. This would answer also https://github.com/corona-warn-app/cwa-wishlist/issues/726, correct @Ein-Tim?

On a side note, you might be interested in this little script, which shows all the booster notifications in one go. Works in a bash shell (tested on Mac):

```
for i in $(wget -qO- https://distribution.dcc-rules.de/bnrules | jq '.[].hash' | tr -d '"')
do
   echo $i 
   wget -qO-  https://distribution.dcc-rules.de/bnrules/$i | jq .
done
 ```

**Comment by Ein-Tim** [comment]
> This would answer also https://github.com/corona-warn-app/cwa-wishlist/issues/726, correct @Ein-Tim?

No @dsarkar. https://github.com/corona-warn-app/cwa-wishlist/issues/726 is about an upcoming change regarding the length of validity for vaccination certificates (9 months). Maybe you could mirror https://github.com/corona-warn-app/cwa-wishlist/issues/726 to JIRA so that the team can decide what to do?

**Comment by MikeMcC399** [comment]
@dsarkar 
Thank you for checking up. I don't understand why the booster rules don't seem to  reflect the published recommendations of STIKO, however I don't know what the organizational responsibilities and process to create and maintain the rules look like so I am quite in the dark.

Thank you also for the script! 🚀 I was able to run it under Ubuntu. That is much easier than viewing the rules manually!

**Comment by dsarkar** [comment]
@MikeMcC399 @Ein-Tim A booster notification is in preparation. No ETA so far.

**Comment by MikeMcC399** [comment]
@dsarkar 
Is there any update on the BNR-DE-0300 rule (see https://github.com/corona-warn-app/cwa-documentation/issues/754#issuecomment-999418904). It seems there is still a bug in this rule restricting it to >= 70 year old people.

There are multi-channel communications through the press (print medien, television, etc.) advising more or less everybody to get a booster vaccination, however it seems strange that the CWA booster notification looks wrong and hasn't been corrected. Maybe I'm missing something here?

**Comment by MikeMcC399** [comment]
When I reviewed the repro steps of issue https://github.com/corona-warn-app/cwa-documentation/issues/710 I used the test certificate https://github.com/eu-digital-green-certificates/dgc-testdata/blob/main/DE/2DCode/png/1.png with parameters https://github.com/eu-digital-green-certificates/dgc-testdata/blob/main/DE/2DCode/raw/1.json

Erika Mustermann was born on August 12, 1964 (making her 58 years of age) and she was vaccinated 2/2 on May 29, 2021 (which is more than 6 months ago).

This certificate fired the rule BNR-DE-0300, recommending  a booster shot, with the information "Last vaccination 222 days ago".

So apparently either:
- the rule on https://distribution.dcc-rules.de/bnrules/cd7defef43c2c0edb115090e669c2bb9028608e247c71b3d0f44728a3df4126f is not the BNR-DE-0300 being used
- the restriction about > 70 years of age is not being applied
- there is something else I did not understand

**Comment by MikeMcC399** [comment]
I made a mistake reading the rules!

- `BNR-DE-0100` is for people older than 70 years old (`not-before plusTime 70 year`)
- `BNR-DE-0300` is for people younger than 70 years old (`before plusTime 70 year`)

If the condition were not applied to the `BNR-DE-0300` rule then people older than 70 years would get that notification in addition to getting the `BNR-DE-0100` notification. They should only get one notification, not two, so this all makes sense (finally!).

**Comment by Ein-Tim** [comment]
FYI: https://tsde.li/stiko-booster-ab-12 says: "Auch 12- bis 17-Jährige sollen nach Ansicht der Ständigen Impfkommission eine Corona-Auffrischungsimpfung erhalten."

**Comment by MikeMcC399** [comment]
@Ein-Tim 

In the text you quote it also says:

"Der Beschlussentwurf der Empfehlung muss nun noch in ein sogenanntes Stellungnahmeverfahren mit den Bundesländern und beteiligten Fachkreisen, ist also noch keine endgültige Empfehlung."

**Comment by Ein-Tim** [comment]
@MikeMcC399 

If you read through the OP, you'll see that this was also the case for the recommendation of booster for everyone above 18, in the end we now have this rule.

I don't think something other will happen here, this recommendation will come sooner or later. 

In general it seems like atm the BNRs are unmaintained, so it's not very important whether this specific rules is already an official suggestion or not.

**Comment by MikeMcC399** [comment]
To be concrete here in this question thread:

The blog announcement [Corona-Warn-App notifies users about a booster vaccination](https://www.coronawarn.app/en/blog/2021-11-04-cwa-booster-notification/) from Nov 4, 2021 says, "The Corona-Warn-App will soon notify the first users for whom the Standing Committee on Vaccination (STIKO) recommends a booster vaccination." and links to More information on the STIKO recommendations can be found on the [RKI's website](https://www.rki.de/DE/Content/Infekt/Impfen/ImpfungenAZ/COVID-19/Impfempfehlung-Zusfassung.html).

It says there:

"**Beschluss der STIKO zur 16. Aktualisierung der COVID-19-Impfempfehlung**

Die STIKO ändert ihre Empfehlung zur COVID-19-Auffrischimpfung hinsichtlich des Impfabstandes. Wie das Epidemiologische Bulletin 2/2022 ausführt, kann die Auffrischimpfung mit einem mRNA-Impfstoff für Personen ≥18 Jahre bereits ab dem vollendeten 3. Monat nach Abschluss der Grundimmunisierung verabreicht werden. Personen, die eine labordiagnostisch gesicherte SARS-CoV-2-Infektion durchgemacht haben, sollen bis auf weiteres eine einmalige COVID-19-Impstoffdosis im Abstand von mindestens 3 Monaten zur Infektion erhalten. Wegen des höheren Risikos für einen schweren COVID-19-Verlauf sollen ältere oder vorerkrankte Personen bei den Auffrischimpfungen unbedingt bevorzugt berücksichtigt werden.

Dieser Artikel wurde am 21.12.2021 online vorab veröffentlicht."

The [Epidemiologisches Bulletin 2 2022](https://www.rki.de/DE/Content/Infekt/EpidBull/Archiv/2022/Ausgaben/02_22.pdf?__blob=publicationFile) was published yesterday, Jan 13, 2022.

So the question is, given that there was a formal announcement of change to the official recommendations on Dec 21, 2021, which was officially confirmed on Jan 13, 2022, why are these rules not reflected in CWA at this time?

I cannot see any corresponding change to the rules on https://distribution.dcc-rules.de/bnrules.

**Comment by MikeMcC399** [comment]
- The concrete case of the 16th STIKO recommendation not being used in the CWA is described in a separate bug report #806. This is about a specific state of the rules which needs to be changed. If and when they are changed, then the bug report can be closed.

- The bug report is separate to not confuse it with the general maintenance of the rules according to each time new recommendations are published, which is more a process question. This question could be closed if there is a description of the process and a commitment to maintain the rules in an on-going way.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

Here we go:

Regarding https://github.com/corona-warn-app/cwa-documentation/issues/754#issuecomment-1012504540, https://www.rki.de/DE/Content/Infekt/EpidBull/Archiv/2022/03/Art_02.html is now the official announcement of the new recommendation.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

There is a typo in the title
"send" => "sent"

**Comment by Ein-Tim** [comment]
Thank you @MikeMcC399!

**Comment by dsarkar** [comment]
@MikeMcC399  @Ein-Tim Currently we do not have information when the BNR will be updated. The question has been forwarded.

**Comment by MikeMcC399** [comment]
@dsarkar 
> Currently we do not have information when the BNR will be updated. The question has been forwarded.

Thank you for forwarding!

The decision-making has been quite untransparent on this subject so far. The response in https://github.com/corona-warn-app/cwa-documentation/issues/814#issuecomment-1044374018 basically only said to be patient, which provided little help on understanding the content or time-lines of any planning.

**Comment by Ein-Tim** [comment]
FYI: https://github.com/corona-warn-app/dcc-rule-translation/issues/4#issuecomment-1066888251 from @BrittaTSI says:

> As you know, the recommendations of RKI/BMG have changed with regards to time frames and age groups since we had first defined these rules. So for BNR 100/110/200/210 there are updates planned that will probably be released in synch with CWA R2.21.

**Comment by MikeMcC399** [comment]
There was another update published today. See https://www.rki.de/DE/Content/Infekt/Impfen/ImpfungenAZ/COVID-19/Impfempfehlung-Zusfassung.html

"Am 31.3.2022 ist die 19. Aktualisierung der COVID-19-Impfempfehlung erschienen. Darin empfiehlt die STIKO Personen, die mit einem nicht in der EU zugelassenen inaktivierten Ganzvirusimpfstoff (CoronaVac von Sinovac, Covilo von Sinopharm oder Covaxin von Bharat Biotech International Ltd.) oder mit dem Vektor-basierten Impfstoff Sputnik V von Gamelaya geimpft wurden, zur Optimierung ihres Impfschutzes eine Auffrischimpfung mit einem mRNA-Impfstoff nach den bestehenden STIKO-Empfehlungen."

I'm sure nobody expects CWA to implement the recommendations in real time, however the booster rules used by CWA seem to be many months behind the official recommendations, which lessens the value of the app in this area.

**Comment by Ein-Tim** [comment]
This PR updates the BNR: https://github.com/corona-warn-app/dcc-rule-translation/pull/6

Booster Notification Rule `3000` should be deleted, as there are no recommendations for boosters from the BMG anymore.

**Comment by BrittaTSI** [comment]
> There was another update published today. See https://www.rki.de/DE/Content/Infekt/Impfen/ImpfungenAZ/COVID-19/Impfempfehlung-Zusfassung.html
> 
> "Am 31.3.2022 ist die 19. Aktualisierung der COVID-19-Impfempfehlung erschienen. Darin empfiehlt die STIKO Personen, die mit einem nicht in der EU zugelassenen inaktivierten Ganzvirusimpfstoff (CoronaVac von Sinovac, Covilo von Sinopharm oder Covaxin von Bharat Biotech International Ltd.) oder mit dem Vektor-basierten Impfstoff Sputnik V von Gamelaya geimpft wurden, zur Optimierung ihres Impfschutzes eine Auffrischimpfung mit einem mRNA-Impfstoff nach den bestehenden STIKO-Empfehlungen."
> 
> I'm sure nobody expects CWA to implement the recommendations in real time, however the booster rules used by CWA seem to be many months behind the official recommendations, which lessens the value of the app in this area.

We are currently working on an update of the rules that will reflect the recommendation of a booster shot for all persons older than 12 years and 3 or more moths after a complete vaccination cycle.
BMG and RKI will decide about the necessity of further updates. 
We will provide the rules in more languages. Now we have DE and EN. TR, RO, BG, PL and possibly UKR will be added.

**Comment by BrittaTSI** [comment]
> This PR updates the BNR: [corona-warn-app/dcc-rule-translation#6](https://github.com/corona-warn-app/dcc-rule-translation/pull/6)
> 
> Booster Notification Rule `3000` should be deleted, as there are no recommendations for boosters from the BMG anymore.

You are right. BNR-DE-300 will be deleted as soon as the updates of the other rules are complete.

**Comment by MikeMcC399** [comment]
@BrittaTSI 

Are you planning to also use booster notifications to warn about the possible loss of basic immunization status after Sep 30, 2022? See https://github.com/corona-warn-app/cwa-wishlist/issues/820 for details.

**Comment by Ein-Tim** [comment]
Updated BNR are live, see e.g. https://distribution.dcc-rules.de/bnrules/74637cc4a9ba588a25ba14e029a232c0a18ea69858c98115c03e60470aa26bd4


<img src="https://user-images.githubusercontent.com/67682506/166712563-e88bce55-5e67-47d7-b3b0-e26efbbfdbdb.png" width="300">


Closing this as answered/implemented.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

I still receive BNR-DE-0300 with a 2/2 vaccination certificate, although according to https://github.com/corona-warn-app/cwa-documentation/issues/754#issuecomment-1088591315 it was supposed to be deleted.
